rbac/rbac/middleware.py

```diff
@@ -231,6 +233,12 @@ def process_request(self, request):  # pylint: disable=R1710
                 user.user_id = None
                 user.system = False
 
+                # The requests made by service accounts are expected to come with an Authorization header which
+                # contains a Bearer token. Therefore, we will attempt to extract it and validate it, and also store it
+                # in case we need to use it to contact IT with it.
+                token_validator = ITSSOTokenValidator()
+                user.bearer_token = token_validator.validate_token(request=request)
```

Comment:
If we put the exception catch around here, might be easier to locate. But as you put currently, the code is neat.
I have no strong preference

Reply:
True, yeah. I think I also prefer having them closer to the function that might raise them. Refactored!